bundles/admin/admin-announcements/publisher/AnnouncementsTool.js

```diff
@@ -0,0 +1,349 @@
+                    content.find('div.ann-title').append(announcementInput);
+                }
+            });
+
+            // Announcement is selected
+            content.find('input[name=announcement]').on('change', function () {
+                var announcement = me.announcements.find(item => item.title === jQuery(this).val());
+                // check if announcement is already checked, if is, add/remove accordingly
+                if (!this.checked) {
+                    me.selectedAnnouncements = me.selectedAnnouncements.filter(function (ann) {
+                        return ann.title !== announcement.title;
+                    });
+                } else {
+                    me.selectedAnnouncements.push(announcement);
+                }
+                me.getPlugin().updateAnnouncements(me.selectedAnnouncements);
+
+                // Shows user the currently selected announcement titles next to the tool (informative input/non-functional)
+                jQuery('div.basic_publisher').find('input[name=publisher-announcements]').val(me.selectedAnnouncements.map(i => i.title).toString());
+            });
+
+            popup.show(title, content, [closeButton]);
+            me.announcementsPopup = popup;
+        },
+
+        /**
+         * Check if Announcement is inside the given timeframe
+         * @method @private isAnnouncementValid
+         * @param  {Object} announcement announcement
+         * @return {Boolean} true if announcement is valid
+         */
+        isAnnouncementValid: function (announcement) {
+            const announcementEnd = new Date(announcement.end_date);
+            const currentDate = new Date();
+            return currentDate.getTime() <= announcementEnd.getTime() ? true : false;
+        },
+
+        /**
+         * Should preselect announcement for tool popup.
+         * @method @private shouldPreselectAnnouncement
+         * @param  {Integer} id announcement id
+         * @return {Boolean} true if announcement must be preselect
+         */
+        shouldPreselectAnnouncement: function (announcement) {
+            const toolPluginAnnouncementsConf = this._getToolPluginAnnouncementsConf();
+            if (toolPluginAnnouncementsConf) {
+                var isPluginConfig = !!((toolPluginAnnouncementsConf && toolPluginAnnouncementsConf.config &&
+                    toolPluginAnnouncementsConf.config.announcements));
+
+                if (isPluginConfig) {
+                    for (var i = 0; i < toolPluginAnnouncementsConf.config.announcements.length; i++) {
+                        if (toolPluginAnnouncementsConf.config.announcements[i] === announcement.id) {
+                            return true;
+                        }
+                    }
+                } else {
+                    return false;
+                }
+            } else {
+                return false;
+            }
```

Comment:
Something like this would make it more readable:
```suggestion
            if (!toolPluginAnnouncementsConf || !toolPluginAnnouncementsConf.config) {
                return false;
            }
            const announcementSelection = toolPluginAnnouncementsConf.config.announcements;
            if (!Array.isArray(announcementSelection)) {
                return false;
            }
            return announcementSelection.includes(announcement.id)
```